source/adios2/core/IO.cpp

```diff
@@ -45,6 +45,8 @@ void IO::SetIOMode(const IOMode ioMode) { m_IOMode = ioMode; };
 
 void IO::SetParameters(const Params &parameters) { m_Parameters = parameters; }
 
+const Params &IO::GetParameters() { return m_Parameters; }
```

Comment:
It's part of the return type, i.e. const ref, so it forces you to treat it as such.  For instance:

```
class Bar
{
  Foo m_Foo;
  const Foo& GetFoo() const { return m_Foo};
};
const Foo& foo1 = Bar.GetFoo(); // fine
Foo& foo2 = Bar.GetFoo(); // Error
```
The lvalue of GetFoo cannot be assigned to a mutable reference, it has to be const, which forces const access to the m_Foo member, even as it's reference is exposed publicly.

Also verified no warnings with Intel 17 compiler.